src/clock.rs

```diff
@@ -39,7 +38,14 @@ impl Instant {
             (
                 InstantKind::Virtual { nanoseconds },
                 InstantKind::Virtual { nanoseconds: earlier },
-            ) => Duration::from_nanos(nanoseconds.saturating_sub(earlier)),
+            ) => {
+                let duration = nanoseconds.saturating_sub(earlier);
+                // It is possible for second to overflow because u64::MAX < (u128::MAX / 1e9).
```

Comment:
```suggestion
                // `Duration` does not provide a nice constructor from a `u128` of nanoseconds,
                // so we have to implement this ourselves.
                // It is possible for second to overflow because u64::MAX < (u128::MAX / 1e9).
```